Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/DriverOperatorAssociationExtra.hs

```diff
@@ -74,6 +74,25 @@ checkDriverOperatorAssociation driverId operatorId = do
     Nothing -> return False
     Just driverOperatorAssociation -> return (isJust driverOperatorAssociation.onboardingVehicleCategory)
 
+findByDriverId ::
+  (EsqDBFlow m r, MonadFlow m, CacheFlow m r) =>
+  Id DP.Person ->
+  Bool ->
+  m (Maybe DriverOperatorAssociation)
+findByDriverId driverId isActive = do
+  now <- getCurrentTime
+  listToMaybe
+    <$> findAllWithOptionsKV
+      [ Se.And
+          [ Se.Is BeamDOA.driverId $ Se.Eq (driverId.getId),
+            Se.Is BeamDOA.isActive $ Se.Eq isActive,
+            Se.Is BeamDOA.associatedTill (Se.GreaterThan $ Just now)
+          ]
+      ]
+      (Se.Desc BeamDOA.createdAt)
+      (Just 1)
+      Nothing
```

Comment:
this change required for `postOperatorConsent` api, because potentially driver can have more than one inactive association, and we need to fetch latest

```
postOperatorConsent (mbDriverId, merchantId, merchantOperatingCityId) = do
  driverId <- fromMaybeM DriverNotFoundWithId mbDriverId
  driver <- QPerson.findById driverId >>= fromMaybeM (PersonNotFound driverId.getId)
  driverOperatorAssociation <- QDriverOperatorAssociation.findByDriverId driverId False >>= fromMaybeM (InactiveOperatorDriverAssociationNotFound driverId.getId)
```